pyzeebe/worker/task_router.py

```diff
@@ -21,25 +29,60 @@ def __init__(self, before: List[TaskDecorator] = None, after: List[TaskDecorator
         self._after: List[TaskDecorator] = after or []
         self.tasks: List[Task] = []
 
-    def task(self, task_config: Union[TaskConfig, str]):
+    def task(self, task_type: str, exception_handler: ExceptionHandler = default_exception_handler,
+             variables_to_fetch: Optional[List[str]] = None, timeout_ms: int = 10000, max_jobs_to_activate: int = 32,
+             before: List[TaskDecorator] = None, after: List[TaskDecorator] = None, single_value: bool = False,
+             variable_name: str = None):
         """
         Decorator to create a task
-
         Args:
-            task_config (Union[str, TaskConfig]): Either the task type or a task configuration object
+            task_type (str): The task type
+
+            exception_handler (ExceptionHandler): Handler that will be called when a job fails. 
+
+            variables_to_fetch (Optional[List[str]]): The variables to request from Zeebe when activating jobs.
+
+            timeout_ms (int): Maximum duration of the task in milliseconds. If the timeout is surpassed Zeebe will give up 
+                                on the worker and retry it. Default: 10000 (10 seconds).
 
+            max_jobs_to_activate (int):  Maximum jobs the worker will execute in parallel (of this task). Default: 32
+
+            before (List[TaskDecorator]): All decorators which should be performed before the task.
+
+            after (List[TaskDecorator]): All decorators which should be performed after the task.
+
+            single_value (bool): If the function returns a single value (int, string, list) and not a dictionary set
+                                 this to True. Default: False
+
+            variable_name (str): If single_value then this will be the variable name given to zeebe:
+                                        { <variable_name>: <function_return_value> }
         Raises:
             DuplicateTaskTypeError: If a task from the router already exists in the worker
-
+            NoVariableNameGivenError: When single_value is set, but no variable_name is given
         """
-        if isinstance(task_config, str):
-            task_config = TaskConfig(task_config)
-        config_with_decorators = self._add_decorators_to_config(task_config)
-
-        def task_wrapper(fn: Callable):
-            task = task_builder.build_task(fn, config_with_decorators)
+        if single_value and not variable_name:
```

Comment:
Can we move back this check to `TaskConfig.__init__`, and build the `TaskConfig` outside of the wrapper? It seems to belong there.

Reply:
The task config is built here because we need the `task_function` to get `variables_to_fetch`